### ISSUE TITLE
Exporting fast.ai ResNet to ONNX and TRT

### ISSUE THREAD

**Issue by mfruhner** [onnx/onnx-tensorrt#545]
Hi,
I trained a ResNet 18 with 6 classes using fast.ai v2 and it works really great so far.
Now I would like to export it, so I can use it on a Nvidia Jetson.

I don’t want to install fastai on the Jetson, as I is quite complex as I found out.
So I exported the model to ONNX with

`torch.onnx.export()`

Then I try to convert it to a TRT engine right on the Jetson with onnx2trt, but I get the following error:

```
Input filename:   opset11.onnx
ONNX IR version:  0.0.6
Opset version:    11
Producer name:    pytorch
Producer version: 1.7
Domain:
Model version:    0
Doc string:
----------------------------------------------------------------
Parsing model
[2020-10-20 09:49:21 WARNING] [TRT]/home/nvidia/git/onnx-tensorrt/onnx2trt_utils.cpp:220: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[2020-10-20 09:49:21   ERROR] (Unnamed Layer* 76) [Shuffle]: at most one dimension may be inferred
While parsing node number 75 [BatchNormalization -> "210"]:
ERROR: /home/nvidia/git/onnx-tensorrt/onnx2trt_utils.cpp:1498 In function scaleHelper:
[8] Assertion failed: dims.nbDims == 4 || dims.nbDims == 5
```

Does this mean, that some operations are not supported?
It does not give much info to me.

Do you have any suggestions?

Thanks!

**Comment by hewumars** [comment]
I have the same problem.
> Input filename:   veid_resnet34.onnx
ONNX IR version:  0.0.6
Opset version:    12
Producer name:    pytorch
Producer version: 1.6
Domain:
Model version:    0
Doc string:
[10/21/2020-16:27:06] [W] [TRT] onnx2trt_utils.cpp:220: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[10/21/2020-16:27:06] [E] [TRT] (Unnamed Layer* 155) [Shuffle]: at most one dimension may be inferred
ERROR: onnx2trt_utils.cpp:1498 In function scaleHelper:
[8] Assertion failed: dims.nbDims == 4 || dims.nbDims == 5
[10/21/2020-16:27:06] [E] Failed to parse onnx file
[10/21/2020-16:27:06] [E] Parsing model failed
[10/21/2020-16:27:06] [E] Engine creation failed
[10/21/2020-16:27:06] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec # F:\HeWu_WorkSpace\Linux\NVIDIA\TensorRT-7.1.3.4\lib\trtexec.exe --onnx=veid_resnet34.onnx --explicitBatch

**Comment by hewumars** [comment]
it works. I solved it by t.view(b, -1) or t.reshape(b, -1)  change to torch.flatten(t, 1).

**Comment by mfruhner** [comment]
Hey, thanks for your answer. Where exactly did you change that? In the fast.ai model code? Or in the converter?

**Comment by kevinch-nv** [comment]
@mfruhner is it possible for you to provide your ONNX model to us to investigate? It looks like a pytorch2onnx export pattern that's generating an ONNX graph that's incompatible with TensorRT.

**Comment by mfruhner** [comment]
Hey,
here is a [OneDrive Link](https://hsosnabrueck-my.sharepoint.com/:u:/g/personal/m_fruhner_hs-osnabrueck_de/EVWB3MWiDBFHjCejujSglaYB0cX2trmsOWerN9KUyzhEBw?e=ribf0I) to my model. It is basically a fast.ai ResNet 18 exported to onnx from the fast.ai learner.

Thanks for your help!

**Comment by FamousDirector** [comment]
I think I am having a similar issue: https://github.com/onnx/onnx-tensorrt/issues/566

I am also converting from a Pytorch model to ONNX then to TensorRT. This model converted fine in the previous 7.1 release.

**Comment by phamdat09** [comment]
> it work. I solved it by t.view(b, -1) or t.reshape(b, -1) change to torch.flatten(t, 1).
It does not seem to work in my case! Did you change something esle ?? Thanhk

**Comment by mfruhner** [comment]
I tried again with the latest commit and it seems to work now, thanks!
